frontend/src/pages/clusterSettings/ClusterSettings.tsx

```diff
@@ -101,42 +114,64 @@ const ClusterSettings: React.FC = () => {
       },
       modelServingPlatformEnabled: modelServingEnabledPlatforms,
     };
-    if (!_.isEqual(clusterSettings, newClusterSettings)) {
-      if (
-        Number(newClusterSettings.pvcSize) !== 0 &&
-        Number(newClusterSettings.cullerTimeout) >= MIN_CULLER_TIMEOUT
-      ) {
-        setSaving(true);
-        updateClusterSettings(newClusterSettings)
-          .then((response) => {
-            setSaving(false);
-            if (response.success) {
-              setClusterSettings(newClusterSettings);
-              dispatch(
-                addNotification({
-                  status: AlertVariant.success,
-                  title: 'Cluster settings changes saved',
-                  message: 'It may take up to 2 minutes for configuration changes to be applied.',
-                  timestamp: new Date(),
-                }),
-              );
-            } else {
-              throw new Error(response.error);
-            }
-          })
-          .catch((e) => {
-            setSaving(false);
-            dispatch(
-              addNotification({
-                status: AlertVariant.danger,
-                title: 'Error',
-                message: e.message,
-                timestamp: new Date(),
-              }),
-            );
-          });
-      }
+
+    const clusterSettingsUnchanged = _.isEqual(clusterSettings, newClusterSettings);
+    const instructLabUnchanged = instructLabEnabled === instructLabDSCState;
+
+    if (clusterSettingsUnchanged && instructLabUnchanged) {
+      return;
     }
+
+    if (
+      Number(newClusterSettings.pvcSize) === 0 ||
+      Number(newClusterSettings.cullerTimeout) < MIN_CULLER_TIMEOUT
+    ) {
+      return;
+    }
+
+    setSaving(true);
+
+    const clusterSettingsPromise = clusterSettingsUnchanged
+      ? Promise.resolve()
+      : updateClusterSettings(newClusterSettings).then((response) => {
+          if (!response.success) {
+            throw new Error(response.error);
+          }
+          setClusterSettings(newClusterSettings);
+        });
+
+    const instructLabPromise = instructLabUnchanged
+      ? Promise.resolve()
+      : toggleInstructLabState(instructLabEnabled ? 'Managed' : 'Removed', 'default-dsc').then(
```

Comment:
dont hard code that. you need to fetch for the dcs, get the name, then use that.